launch/test/launch/actions/test_log.py

```diff
@@ -81,3 +81,11 @@ def test_log_execute():
     log = Log(msg='foo', level='ERROR')
     launch_context = LaunchContext()
     assert log.visit(launch_context) is None
+
+
+def test_log_level_error():
+    """Checks for error message to be raised given invalid level."""
+    try:
+        Log(msg='foo', level='foo')
+    except KeyError as e:
+        assert 'Invalid log level ' in str(e)
```

Comment:
This doesn't actually assert that an error is raised. You can use `with pytest.raises(KeyError, match=r'Invalid log level.*'):`: https://docs.pytest.org/en/stable/how-to/assert.html#matching-exception-messages

And an error wouldn't be raised in this case if you're just calling the constructor; you have to call `visit()` on it so that `execute()` is called.